src/StringHelper.php

```diff
@@ -429,4 +429,37 @@ public static function htmlspecialchars(string $string, int $flags, string $enco
             ? htmlspecialchars($string, $flags)
             : htmlspecialchars($string, $flags, $encoding ?: ini_get('default_charset'), $double_encode);
     }
+    
+    /**
+     * Performs string comparison using timing attack resistant approach.
+     * @see http://codereview.stackexchange.com/questions/13512
+     * @param string $expected string to compare.
+     * @param string $actual user-supplied string.
+     * @return bool whether strings are equal.
+     */
+    public function compareString(string $expected, string $actual): bool
+    {
+        if (!is_string($expected)) {
```

Comment:
why duplicate the check if the arguments are specified that only a string accepts?

Reply:
Oh, you are right